### PR TITLE
Catch more exceptions in jsonCompiler.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/970

### CODE REVIEW

File: solc/jsonCompiler.cpp [hunk 1]
```diff
@@ -203,51 +203,65 @@ string compile(StringMap const& _sources, bool _optimize, CStyleReadFileCallback
...
+				contractData["srcmapRuntime"] = runtimeSourceMap ? *runtimeSourceMap : "";
+				ostringstream unused;
+				contractData["assembly"] = compiler.streamAssembly(unused, contractName, _sources, true);
+				output["contracts"][contractName] = contractData;
+			}
+
+			// Do not taint the internal error list
+			ErrorList formalErrors;
+			if (compiler.prepareFormalAnalysis(&formalErrors))
+				output["formal"]["why3"] = compiler.formalTranslation();
```
Comment (axic) [inline solc/jsonCompiler.cpp:230]: I think this should have its own try/catch block after the EVM compiler. Failing why3 could still result in a working EVM compilation and with the current change that will block it as well.

Comment (chriseth) [reply solc/jsonCompiler.cpp:230]: While I agree, the fact that `formalTranslation` leaks exceptions is still a serious compiler bug. If we catch it here and convert it to an error message in the output, nobody will notice such things.

Comment (pirapira) [reply solc/jsonCompiler.cpp:230]: @chriseth if we separate the try/catch block around JSON generation and the one around why3 generation, the chance is higher that the problem is recognized correctly.

`Unknown exception while generating Why3 output.` would be more informative than `Unknown exception while generating compiler output.`.

Comment (chriseth) [reply solc/jsonCompiler.cpp:230]: Sure, but with the same argument, we could add a try/catch block around every single line. Compiler bugs are easy to reproduce and a debugger will show you the exact position of the problem. The benefit to the user could be to switch off the problematic part in the output, but so far this has only happened once.

Comment (pirapira) [reply solc/jsonCompiler.cpp:230]: I'm fine to merge whichever ways (unified or separate).

Is there a standard test of regression tests that should be run before every merge?

**Pull request merged**